lib/styles.js

```diff
@@ -347,4 +347,118 @@ module.exports = function (expect) {
+        var i;
+        this.nl(options.top || 0).sp(options.left || 0).text('┌', styles);
+        for (i = 1 ; i < options.width ; i += 1) {
+            this.text(i === options.width - 1 && options.direction === 'up' ? '▷' : '─', styles);
+        }
+        this.nl();
+        for (i = 1 ; i < options.height - 1 ; i += 1) {
+            this.sp(options.left || 0).text('│', styles).nl();
+        }
+        this.sp(options.left || 0).text('└', styles);
+        for (i = 1 ; i < options.width ; i += 1) {
+            this.text(i === options.width - 1 && options.direction === 'down' ? '▷' : '─', styles);
+        }
+    });
+
+    var flattenBlocksInLines = require('magicpen/lib/flattenBlocksInLines');
+    expect.addStyle('merge', function (pens) {
+        var flattenedPens = pens.map(function (pen) {
+            return flattenBlocksInLines(pen.output);
+        }).reverse();
+        var maxHeight = flattenedPens.reduce(function (maxHeight, pen) {
+            return Math.max(maxHeight, pen.length);
+        }, 0);
+        var blockNumbers = new Array(flattenedPens.length);
+        var blockOffsets = new Array(flattenedPens.length);
+        // As long as there's at least one pen with a line left:
+        for (var lineNumber = 0 ; lineNumber < maxHeight ; lineNumber += 1) {
+            if (lineNumber > 0) {
+                this.nl();
+            }
+            var i;
+            for (i = 0 ; i < blockNumbers.length ; i += 1) {
+                blockNumbers[i] = 0;
+                blockOffsets[i] = 0;
+            }
+            var contentLeft;
+            do {
+                contentLeft = false;
+                var hasOutputChar = false;
+                for (i = 0 ; i < flattenedPens.length ; i += 1) {
+                    var currentLine = flattenedPens[i][lineNumber];
+                    if (currentLine) {
+                        while (currentLine[blockNumbers[i]] && blockOffsets[i] >= currentLine[blockNumbers[i]].args.content.length) {
+                            blockNumbers[i] += 1;
+                            blockOffsets[i] = 0;
+                        }
+                        var currentBlock = currentLine[blockNumbers[i]];
+                        if (currentBlock) {
+                            contentLeft = true;
+                            if (!hasOutputChar) {
+                                var ch = currentBlock.args.content.charAt(blockOffsets[i]);
+                                if (ch !== ' ') {
+                                    this.text(ch, currentBlock.args.styles);
+                                    hasOutputChar = true;
+                                }
+                            }
+                            blockOffsets[i] += 1;
+                        }
+                    }
+                }
+                if (!hasOutputChar && contentLeft) {
+                    this.sp();
+                }
+            } while (contentLeft);
+        }
+    });
```

Comment:
As we talked about, this style should move to magicpen core when it has proven it's stability.